salt/addons/init.sls

```diff
@@ -14,40 +10,44 @@ include:
     - source:      salt://addons/addons/namespace.yaml.jinja
     - template:    jinja
 
-{% if salt['pillar.get']('addons:dns', 'false').lower() == 'true' %}
-/etc/kubernetes/addons/kubedns-sa.yaml:
-  file.managed:
-    - source:     salt://addons/addons/kubedns-sa.yaml.jinja
-    - template:   jinja
-
-/etc/kubernetes/addons/kubedns-cm.yaml:
-  file.managed:
-    - source:      salt://addons/addons/kubedns-cm.yaml.jinja
-    - template:    jinja
+apply-namespace:
+  cmd.run:
+    - name: |
+        kubectl apply -f /etc/kubernetes/addons/namespace.yaml
```

Comment:
Actually, this makes a lot of sense. We (briefly) chatted about this on IRC earlier and I had suggested either the admin node (which is actually much harder than it should be), or salts `subset: 1` matcher, which we don't have in our version of salt.

The salt master container could easily be setup to handle these "once per cluster" kind of tasks, is easily handled during upgrades (unlike the admin node), and doesn't add more responsibility onto the admin node - because we want to, at some point, stop wasting a hardware node.

But, yes, I agree with Rob that doing this it's it's own topic, something that should be looked at in the context of all the "once per cluster" things we do (etcd discovery setup, flannel setup, probably more) and designed, rather than just added as part of a random PR to cover just 1 use case.